library/src/main/java/com/getstream/sdk/chat/utils/UploadManager.java

```diff
@@ -0,0 +1,101 @@
+package com.getstream.sdk.chat.utils;
+
+import com.getstream.sdk.chat.R;
+import com.getstream.sdk.chat.StreamChat;
+import com.getstream.sdk.chat.model.Attachment;
+import com.getstream.sdk.chat.model.Channel;
+import com.getstream.sdk.chat.model.ModelType;
+import com.getstream.sdk.chat.rest.interfaces.UploadFileCallback;
+import com.getstream.sdk.chat.rest.response.UploadFileResponse;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UploadManager {
+
+    private static List<Attachment> queue;
+
+    public static void uploadFile(Channel channel, List<Attachment> attachments,
```

Comment:
All these methods shouldn't be static. Static methods are almost only for set of pure util functions.
`UploadManager` should be just a regular class which keeps state inside and doesn't share it staticly.
```kotlin
val uploadManager = new UploadManager()
uploadManager. uploadFile(...)
```